test/Microsoft.AspNet.Mvc.FunctionalTests/InlineConstraintTests.cs

```diff
@@ -408,7 +408,8 @@ public static IEnumerable<object[]> QueryParameters
                         "InlineConstraints_Products",
                         "GetProductById",
                         "id",
-                        "sdsd", ""
+                        "sdsd",
+                        "/area-exists/InlineConstraints_Products/GetProductById?id=sdsd"
```

Comment:
This route is now "greedy". I don't see much reason to try and rationalize this behavior, these tests still test what they did before - that the constraint in question doesn't allow that route to match with invalid values.